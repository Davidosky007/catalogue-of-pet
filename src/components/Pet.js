import React from 'react';
import PropTypes from 'prop-types';
import { Link } from '@reach/router';

const Pet = ({ id, name, media }) => {
  let hero = 'http://placecorgi.com/300/300';

  if (media.length !== 0) {
    hero = media[0].medium;
  }

  return (
    <Link to={`/details/${id}`}>
      <div className="pets__card" data-tesid="pets-comp">
        <div className="pets__card-box">
          <img src={hero} alt="" />
          <span>{name}</span>
        </div>
      </div>
    </Link>
  );
};

Pet.propTypes = {
  name: PropTypes.string,
  media: PropTypes.arrayOf(PropTypes.any),
  id: PropTypes.number,
};

Pet.defaultProps = {
  name: '',
  media: [],
  id: null,
};

export default Pet;
