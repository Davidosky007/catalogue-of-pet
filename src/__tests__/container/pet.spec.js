import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Pet from '../../components/Pet';

test('should render Pets components properly ', () => {
  const details = renderer.create(<Pet />);
  const final = details.toJSON();
  expect(final).toMatchSnapshot();
});
