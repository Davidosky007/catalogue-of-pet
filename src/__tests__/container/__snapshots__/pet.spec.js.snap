// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pets components properly  1`] = `
<a
  href="/details/null"
  onClick={[Function]}
>
  <div
    className="pets__card"
    data-tesid="pets-comp"
  >
    <div
      className="pets__card-box"
    >
      <img
        alt=""
        src="http://placecorgi.com/300/300"
      />
      <span>
        
      </span>
    </div>
  </div>
</a>
`;
